src/netlink_generic.c

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (c) 2023 The strace developers.
+ *
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#include "defs.h"
+#include "netlink.h"
+#include "nlattr.h"
+#include <linux/genetlink.h>
+
+#include "netlink_generic.h"
+#include "xlat/genl_flags.h"
+#include "xlat/genl_nlctrl_types.h"
+#include "xlat/genl_nlctrl_attrs.h"
+#include "xlat/genl_nlctrl_ops_attrs.h"
+#include "xlat/genl_nlctrl_mcast_attrs.h"
+#include "xlat/genl_nlctrl_policy_attrs.h"
+
+static void
+decode_genl_family(struct tcb *const tcp, const uint8_t cmd,
+		   const uint8_t version, const kernel_ulong_t addr,
+		   const unsigned int len)
+{
+	tprint_struct_begin();
+	tprints_field_name("cmd");
+	printxval(NULL, cmd, "???");
+	tprint_struct_next();
+	tprints_field_name("version");
+	tprintf_string("%d", version);
+
+	if (len > 0) {
+		tprint_struct_next();
+		tprints_field_name("data");
```

Comment:
I had studied the way to decode the "data" area.
https://github.com/masatake/strace/commits/netlink-genl

Reply:
I found this is a fallback path. Please, ignore what I wrote. Sorry for making noise.

Reply:
No worries, thanks for checking.